src/utils/Swiper/Swiper.js

```diff
@@ -0,0 +1,84 @@
+export default function Swiper() {
+  const coordinates = {
+    $element: null,
+    xDown: null,
+    yDown: null,
+    xUp: null,
+    yUp: null,
+  };
+
+  const dispatchSwipeLeft = () => {
+    const event = new Event('swipe-left');
+    coordinates.$element.dispatchEvent(event);
+  };
+
+  const dispatchSwipeRight = () => {
+    const event = new Event('swiper-right');
+    coordinates.$element.dispatchEvent(event);
+  };
+
+  const dispatchSwipeUp = () => {
+    const event = new Event('swipe-up');
+    coordinates.$element.dispatchEvent(event);
+  };
+
+  const dispatchSwipeDown = () => {
+    const event = new Event('swipe-down');
+    coordinates.$element.dispatchEvent(event);
+  };
+
+  function swipeDirection() {
+    const xDiff = coordinates.xDown - coordinates.xUp;
+    const yDiff = coordinates.yDown - coordinates.yUp;
```

Comment:
Estou sendo meio chato mas, fique atento que como estamos lidando com a distancia entre dois pontos o correto seria B - A
```suggestion
    const xDiff = coordinates.xUp - coordinates.xDown;
    const yDiff = coordinates.yUp - coordinates.yDown;
```